planner/core/stats.go

```diff
@@ -341,12 +341,15 @@ func (p *LogicalMaxOneRow) DeriveStats(childStats []*property.StatsInfo) (*prope
 // DeriveStats implement LogicalPlan DeriveStats interface.
 func (p *LogicalWindow) DeriveStats(childStats []*property.StatsInfo) (*property.StatsInfo, error) {
 	childProfile := childStats[0]
-	childLen := len(childProfile.Cardinality)
 	p.stats = &property.StatsInfo{
 		RowCount:    childProfile.RowCount,
-		Cardinality: make([]float64, childLen+1),
+		Cardinality: make([]float64, p.schema.Len()),
+	}
+	for i := 0; i < p.schema.Len()-1; i++ {
+		colIdx := p.children[0].Schema().ColumnIndex(p.schema.Columns[i])
```

Comment:
Here we directly visit `p.Children`, which i think is a not compatible operation with current cascades. /cc @eurekaka

Reply:
Yes, cascades planner would panic then. Would this column order change caused by projection elimination impact other codes of window function?

Reply:
Yes, seems after the fix the field info is missing:
```sql
MySQL [test]> select avg(b), max(avg(b)) over() from t group by a;
+--------+--------------------+
|        | max(avg(b)) over() |
+--------+--------------------+
| 1.5000 |             1.5000 |
| 1.5000 |             1.5000 |
+--------+--------------------+
```

Reply:
@eurekaka @lamxTyler The missing field name is fixed in https://github.com/pingcap/tidb/pull/9644.